### PR TITLE
Feature/user controller

### PR DESCRIPTION
**Pull request by idavid80**
- Renaming of the branch: The branch was renamed by removing the capital letters and creating a new version without them (removing the previous version).
- Refactoring of error handling: Ensured proper error handling, avoiding explicit 500 error checks.
- Improved mocks: Improved simulation of password verification using bcrypt to make testing more realistic.
- Added tests for edge cases: Added tests for:
1. Missing required fields.
2. Case sensitivity in user names and other relevant fields.
3. Handling of special characters in entries (usernames, passwords, e-mails).
4. JWT expiry.

### CODE REVIEW

File: src/controllers/user.controller.js [hunk 1]
```diff
@@ -17,7 +27,7 @@ export async function signUp(req, res) {
   if (rows.length > 0) {
     return res.status(400).json({ message: 'Username already exists' });
   }
-  
+  console.log('Password before hashing:', password);
```
Comment (alvarobernal2412) [inline src/controllers/user.controller.js:30]: The console.logs should be deleted before uploading the code, a good trick is to add a // DELETE comment so you can find these code blocks from the editor's search engine.

File: src/controllers/user.controller.js [hunk 2]
```diff
@@ -167,26 +182,31 @@ export async function signOut(req, res) {
...
-  } catch {
-    return res.status(403).json({ message: 'Invalid token' });
+
+    const { authority } = jwt.verify(accessToken, process.env.JWT_SECRET);
+    return res.status(200).json({ authority });
+  } catch (error) {
+    if (error instanceof jwt.TokenExpiredError) {
+      return res.status(401).json({ message: "Token expired" });
+    }
+    return res.status(403).json({ message: "Invalid token" });
```
Comment (alvarobernal2412) [inline src/controllers/user.controller.js:209]: This logic should be avoided as such checks could be done using the verifyAuth middleware, which checks the validity of tokens.

**Review by alvarobernal2412 — CHANGES_REQUESTED** [review]
Great job! A couple of changes and the PR will be ready to merge!

File: src/controllers/user.controller.js [hunk 3]
```diff
@@ -167,26 +178,26 @@ export async function signOut(req, res) {
...
-      .status(401)
-      .json({ message: "No token provided or it's malformed" });
-  } catch {
-    return res.status(403).json({ message: 'Invalid token' });
+    const { authority } = jwt.verify(accessToken, process.env.JWT_SECRET);
+    return res.status(200).json({ authority });
+  } catch (error) {
+    return res.status(403).json({ message: "Invalid token" });
   }
 }
```
Comment (alvarobernal2412) [inline src/controllers/user.controller.js:202]: To maintain proper error handling, you should use the verifyAuth middleware in the getAuthority in the `user.route.js`. This way you can maintain that functionality without duplicating code.

**Review by alvarobernal2412 — CHANGES_REQUESTED** [review]

**Review by alvarobernal2412 — APPROVED** [review]
Great job!
